spring-cloud-data-shell/src/main/java/org/springframework/cloud/data/shell/command/TaskCommands.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.shell.command;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.data.rest.client.TaskOperations;
+import org.springframework.cloud.data.rest.resource.TaskDefinitionResource;
+import org.springframework.cloud.data.rest.util.DeploymentPropertiesUtils;
+import org.springframework.cloud.data.shell.config.CloudDataShell;
+import org.springframework.hateoas.PagedResources;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliAvailabilityIndicator;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.shell.support.table.Table;
+import org.springframework.shell.support.table.TableHeader;
+import org.springframework.stereotype.Component;
+
+/**
+ * Task commands.
+ *
+ * @author Glenn Renfro
+ */
+@Component
+// todo: reenable optionContext attributes
+	public class TaskCommands implements CommandMarker {
+
+	private static final String LIST_TASK = "task list";
+
+	private static final String CREATE_TASK = "task create";
+
+	private static final String LAUNCH_TASK = "task launch";
+
+	private static final String STATUS_TASK = "task status";
+
+	private static final String DESTROY_TASK = "task destroy";
+
+	private static final String PROPERTIES_OPTION = "properties";
+
+	private static final String PROPERTIES_FILE_OPTION = "propertiesFile";
+
+
+	@Autowired
+	private CloudDataShell cloudDataShell;
+
+	@Autowired
+	private UserInput userInput;
+
+	@CliAvailabilityIndicator({ LIST_TASK, CREATE_TASK, LAUNCH_TASK, STATUS_TASK, DESTROY_TASK})
+	public boolean available() {
+		return cloudDataShell.getCloudDataOperations() != null;
+	}
+
+	@CliCommand(value = LIST_TASK, help = "List created tasks")
+	public Table listTasks() {
```

Comment:
I feel like the comment @markfisher made in the `TaskOperations` should be applied here.  We already know these commands are for tasks since it's on the `TaskCommands` class.  We should remove the word `task` from each of these commands.